packages/nimble-components/src/table/specs/table-row-selection.md

```diff
@@ -0,0 +1,136 @@
+    selectedRecordIds: string[];
+}
+```
+
+In the case where grouping is enabled, it is possible that a grouping row can be visualized as selected because all of its child rows are selected. It is important to note that because the grouping row does not have a record associated with it, it will never be returned as part of the `selectedRecordIds` array in the event detail. Instead, each of its selected child rows will have their record ID included in the `selectedRecordIds` array.
+
+The table will have a property called `selectedRecordIds` of type `string[]`. It will be implemented with a getter and setter. The getter will convert TanStack's selection state, which is a map of row IDs to boolean, to a string array of the IDs in the map with a value of `true`. The setter will update TanStack's selection state to be the record IDs provided to the setter. Any record IDs that do not exist will be removed from the selection prior to updating TanStack's state. Setting the `selectedRecordIds` to an empty array will clear the selection.
+
+The table will only support selection when it has an `id-field-name` configured on it. This is for a few reasons:
+
+-   Record IDs are meaningless when an `id-field-name` is not set. When that attribute is not set, the IDs are generated by TanStack, and no assumptions about their values should be made by a client application. Therefore, a client would not be able to do anything with the array of record IDs emitted from the `selection-change` event when `id-field-name` is not set.
+-   Selection within TanStack will be maintained using record IDs. The selection behavior will be unexpected and non-deterministic if a record in the table does not have a static ID associated with it.
+-   The selection state of TanStack needs to be updated when the data is updated to ensure that no rows are configured as selected when they don't exist in the data set anymore. Removing the selection state for removed records will be more straight forward when an `id-field-name` is set because the record IDs can be determined using the data passed to `setData` alone. When using IDs generated by TanStack, first TanStack would need to be updated with the new data, then the generated ID could be pulled from each TanStack row, and only then could the new selection state be determined and set on the underlying TanStack table.
+
+When new data is set on the table, the set of selected rows needs to be evaluated to remove the state of rows whose record no longer exists in the data. When the record IDs are invalid for any reason, the selection state will not be modified based on the new data. When the data updates again to have valid IDs, the selection state will then be updated. When `id-field-name` is changed, the selection state will be cleared (i.e. no rows selected) because the set of old selected record IDs will be invalidated by each record potentially having a new ID.
+
+### General behavior and design
```

Comment:
I thought we had a separate issue for keyboard interaction that covered all keyboard interaction in the table, so I wasn't going to include keyboard interaction as part of this effort. I can't find that issue now. I only see the design issue for it --#877. But, I think it's going to be difficult/impossible to implement keyboard selection until some of that other work is done. For example, if you can't appropriately focus a row in the table, then you can't have keyboard interaction for selecting the focused row.

I can document the desired behavior here, but I'm thinking it will be implemented as part of a separate effort/issue.

Reply:
I created #1137. Documenting the behavior here is nice-to-have but I agree it makes sense to prioritize this new issue separately.